### ISSUE TITLE
[x64_windows,x86_windows] build failure with "A suitable version of python3 was not found"

### ISSUE THREAD

**Issue by SamuelMarks** [microsoft/vcpkg#23295]
**Host Environment**
- OS: Windows
- Compiler: revision

**To Reproduce**
See both x86 and x64 failure logs on https://github.com/microsoft/vcpkg/pull/23234

**Failure logs** 
https://dev.azure.com/vcpkg/public/_build/results?buildId=68053&view=logs&j=878666d5-db33-5b27-9e7d-b0c7ee352005&t=257b0d57-ac73-5c53-4772-914339da5109

**Additional context**
```
2022-02-26T20:04:42.8877903Z ##[section]Starting: Build a file list for all packages
2022-02-26T20:04:42.8967904Z ==============================================================================
2022-02-26T20:04:42.8968386Z Task         : PowerShell
2022-02-26T20:04:42.8968674Z Description  : Run a PowerShell script on Linux, macOS, or Windows
2022-02-26T20:04:42.8968996Z Version      : 2.198.0
2022-02-26T20:04:42.8969267Z Author       : Microsoft Corporation
2022-02-26T20:04:42.8969645Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/powershell
2022-02-26T20:04:42.8970105Z ==============================================================================
2022-02-26T20:04:43.6643447Z Generating script.
2022-02-26T20:04:43.6919825Z ========================== Starting Command Output ===========================
2022-02-26T20:04:43.7097493Z ##[command]"C:\Program Files\PowerShell\7\pwsh.exe" -NoLogo -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command ". 'C:\a\_temp\43b78b7a-6437-419f-aa4a-70b45f5c4770.ps1'"
2022-02-26T20:04:43.9505070Z A suitable version of python3 was not found (required v3.10.1). Downloading portable python3 v3.10.1...
2022-02-26T20:04:43.9505865Z Downloading python3...
2022-02-26T20:04:43.9506353Z   https://www.python.org/ftp/python/3.10.1/python-3.10.1-embed-win32.zip -> D:\downloads\python-3.10.1-embed-win32.zip
2022-02-26T20:04:44.1037508Z Extracting python3...
2022-02-26T20:04:44.1045064Z A suitable version of 7zip was not found (required v19.0.0). Downloading portable 7zip v19.0.0...
2022-02-26T20:04:44.1138831Z Extracting 7zip...
2022-02-26T20:04:44.1561593Z msiexec failed while extracting 'D:\downloads\7z1900-x64.msi' with message:
2022-02-26T20:04:44.1562181Z T
2022-02-26T20:04:44.3572866Z & : The term 'A suitable version of python3 was not found (required v3.10.1). Downloading portable python3 v3.10.1... Extracting python3... A suitable version of 7zip was not found (required v19.0.0). Downloading portable 7zip v19.0.0... Extracting 7zip... msiexec failed while extracting 'D:\downloads\7z1900-x64.msi' with message: T' is not recognized as a name of a cmdlet, function, script file, or executable program.
2022-02-26T20:04:44.3575292Z Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
2022-02-26T20:04:44.3576175Z At C:\a\_temp\43b78b7a-6437-419f-aa4a-70b45f5c4770.ps1:4 char:3
2022-02-26T20:04:44.3579910Z + & $(.\vcpkg fetch python3) .\scripts\file_script.py D:\installed\vcpk …
2022-02-26T20:04:44.3580950Z +   ~~~~~~~~~~~~~~~~~~~~~~~~
2022-02-26T20:04:44.3581753Z + CategoryInfo          : ObjectNotFound: (A suitable version …si' with message: T:String) [], ParentContainsErrorRecordException
2022-02-26T20:04:44.3582559Z + FullyQualifiedErrorId : CommandNotFoundException
2022-02-26T20:04:44.4352934Z ##[error]PowerShell exited with code '1'.
2022-02-26T20:04:44.7251504Z ##[section]Finishing: Build a file list for all packages
```

**Comment by Thomas1664** [comment]
Hi @SamuelMarks 
vcpkg is experiencing CI issues on platform Windows since Friday. Merging master into your PR branch might resolve some of those issues. I posted the issue already here: #23284.
If you look closely at your logs, you will notice that not python3 but 7zip is the problem, although the "fake" python thing is interesting:
```
2022-02-26T20:04:44.1561593Z msiexec failed while extracting 'D:\downloads\7z1900-x64.msi' with message:
2022-02-26T20:04:44.1562181Z T
```

**Comment by SamuelMarks** [comment]
Ok I'll close this issue now since it's being tracked elsewhere. Also you might want to bump 7zip version:
https://www.7-zip.org/a/7z2107-x64.msi where 19.00 was released 2019-02-21 and 21.07 was released 2021-12-26
